### PR TITLE
feat: Support HTTP PROXY

### LINKED ISSUE #193: feat(buildkit): Support build-time HTTP_PROXY
## Description

```
			FrontendAttrs: map[string]string{
				"build-arg:HTTPS_PROXY": os.Getenv("HTTPS_PROXY"),
			},
```

Buildkit supports some build time attrs in its dockerfile frontend. We need to support them in envd.

### PR DESCRIPTION
**Pull request by gaocegege**
A new flag `--use-proxy` is added to inject the environment variables HTTP_PROXY to the build process.

Close #193

### CODE REVIEW

**Comment by kemingy** [comment]
Is it ready to review?

**Comment by gaocegege** [comment]
Yep.

File: pkg/builder/builder.go [hunk 1]
```diff
@@ -346,7 +362,7 @@ func (b generalBuilder) checkIfNeedBuild(ctx context.Context) bool {
 		b.logger.Debugf("failed to check manifest update: %s", err)
 	}
 	if !isUpdated {
-		b.logger.Infof("manifest is not updated, skip building")
+		// b.logger.Infof("manifest is not updated, skip building")
```
Comment (kemingy) [inline pkg/builder/builder.go:365]: Shall we keep this or not?

Comment (gaocegege) [reply pkg/builder/builder.go:365]: It is a typo, thanks for the review. I am reverting it.

**Review by kemingy — APPROVED** [review]

**Pull request merged**